File: C++/009_Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        long long left = 1, right = 10;
        if (x < 0)
            return false;
        if (x == 0)
            return true;
        while (x >= left)
            left *= 10;
        while (left > right) {
            int i = (x % left) * 10 / left;
            int j = (x % right) * 10 / right;
            if (i != j)
                return false;
            left /= 10;
            right *= 10;
        }
        return true;
    }
};

File: Python/009_Palindrome Number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        if x < 10:
            return True
        left = 1
        while left <= x:
            left *= 10
        left /= 10
        right = 1
        while left > right:
            if x // left % 10 != x // right % 10:
                return False
            left, right = left / 10, right * 10
        return True